packages/engine-server/src/util/impl/inMemoryNoteCacheImpl.ts

```diff
@@ -0,0 +1,47 @@
+import { DendronError, NoteProps } from "@dendronhq/common-all";
+import { InMemoryNoteCache } from "../inMemoryNoteCache";
+
+export class InMemoryNoteCacheImpl implements InMemoryNoteCache {
+  /** Note map which maps lowercase file name to a list {@link NoteProps}
+   *  that have the matching file name */
+  private readonly mapFNameToNotes: Map<string, NoteProps[]>;
+
+  static createCache(notes: NoteProps[]) {
```

Comment:
Why static here and not just an instance method + public constructor?  Same with `initializeFileNameMap()` method below.